pyiceberg/manifest.py

```diff
@@ -897,9 +902,11 @@ def prepare_manifest(self, manifest_file: ManifestFile) -> ManifestFile:
 
 class ManifestListWriterV2(ManifestListWriter):
     _commit_snapshot_id: int
-    _sequence_number: int
+    _sequence_number: Optional[int]
```

Comment:
I think this is required for manifest list. This is always known when we write the manifest list because manifest lists are written for every commit attempt (after the sequence number has been updated).

Reply:
I agree, let me check!